pkg/scanners/set_impl_test.go

```diff
@@ -115,12 +115,27 @@ func TestSetOrdering(t *testing.T) {
 			},
 		},
 	}
+
+	scannerFactory.EXPECT().CreateScanner(testutils.PredMatcher("scannerv4", func(integration *storage.ImageIntegration) bool {
+		return integration.GetType() == "scannerv4"
+	})).Return(newFakeImageScanner(&fakeScanner{typ: "scannerv4"}), nil)
+
+	scannerV4Integration := &storage.ImageIntegration{
+		Id:   "scannerv4",
+		Type: "scannerv4",
+		IntegrationConfig: &storage.ImageIntegration_ScannerV4{
+			ScannerV4: &storage.ScannerV4Config{},
+		},
+	}
+
 	require.NoError(t, scannerSet.UpdateImageIntegration(clairifyIntegration))
 	require.NoError(t, scannerSet.UpdateImageIntegration(ecrIntegration))
+	require.NoError(t, scannerSet.UpdateImageIntegration(scannerV4Integration))
 	for i := 0; i < 10000; i++ {
```

Comment:
interesting way to test if this is deterministic

Reply:
If the code is behaving as intended it is deterministic, however if you change (for example) two of the scanners to have the same priority, it's no longer deterministic - and without the iterations the test would become flaky.